src/textbooks/textbook-form/TextbookForm.jsx

```diff
@@ -0,0 +1,192 @@
+import { useState } from 'react';
+import PropTypes from 'prop-types';
+import { useIntl } from '@edx/frontend-platform/i18n';
+import * as Yup from 'yup';
+import { FieldArray, Formik } from 'formik';
+import {
+  PictureAsPdf as PdfIcon,
+  Add as AddIcon,
+  DeleteOutline as DeleteIcon,
+  Upload as UploadIcon,
+} from '@edx/paragon/icons';
+import {
+  ActionRow,
+  Button,
+  Form,
+  Icon,
+  IconButtonWithTooltip,
+  useToggle,
+} from '@edx/paragon';
+
+import FormikControl from '../../generic/FormikControl';
+import PromptIfDirty from '../../generic/PromptIfDirty';
+import ModalDropzone from '../../generic/modal-dropzone/ModalDropzone';
+import { useModel } from '../../generic/model-store';
+import messages from './messages';
+
+const TextbookForm = ({
+  closeTextbookForm,
+  initialFormValues,
+  onSubmit,
+  onSavingStatus,
+  courseId,
+}) => {
+  const intl = useIntl();
+
+  const courseDetail = useModel('courseDetails', courseId);
+  const courseTitle = courseDetail ? courseDetail?.name : '';
+
+  const [currentTextbookIndex, setCurrentTextbookIndex] = useState(0);
+  const [isUploadModalOpen, openUploadModal, closeUploadModal] = useToggle(false);
+
+  const textbookFormValidationSchema = Yup.object().shape({
+    tab_title: Yup.string().required(intl.formatMessage(messages.tabTitleValidationText)).max(255),
+    chapters: Yup.array().of(
+      Yup.object({
+        title: Yup.string().required((intl.formatMessage(messages.chapterTitleValidationText))).max(255),
+        url: Yup.string().required(intl.formatMessage(messages.chapterUrlValidationText)).max(255),
+      }),
+    ).min(1),
+  });
+
+  return (
+    <div className="textbook-form">
+      <Formik
+        initialValues={initialFormValues}
+        validationSchema={textbookFormValidationSchema}
+        onSubmit={onSubmit}
+        validateOnBlur
+        validateOnMount
+      >
+        {({
+          values, handleSubmit, isValid, dirty, setFieldValue,
+        }) => (
+          <>
+            <Form.Group size="sm" className="form-field">
+              <Form.Label size="sm" className="font-weight-bold form-main-label">
+                {intl.formatMessage(messages.tabTitleLabel)} *
+              </Form.Label>
+              <FormikControl
+                name="tab_title"
+                value={values.tab_title}
+                placeholder={intl.formatMessage(messages.tabTitlePlaceholder)}
+              />
+              <Form.Control.Feedback className="form-helper-text">
+                {intl.formatMessage(messages.tabTitleHelperText)}
+              </Form.Control.Feedback>
+            </Form.Group>
+            <FieldArray
+              name="chapters"
+              render={(arrayHelpers) => (
+                <>
+                  {Boolean(values?.chapters.length) && values.chapters.map(({ title, url }, index) => (
+                    <div className="form-chapters-fields" data-testid="form-chapters-fields">
```

Comment:
```suggestion
                    <div key={title} className="form-chapters-fields" data-testid="form-chapters-fields">
```

Reply:
It's not working. title as key will trigger rerender fields

Reply:
Think about how we can best pass the keys for each element in this place (I currently have a proposal to use an index, but I don’t like it)